cg/cli/workflow/balsamic/base.py

```diff
@@ -138,6 +138,9 @@ def run(context, analysis_type, run_analysis, priority, case_id, dry):
         balsamic_analysis_api.balsamic_api.run_analysis(
             arguments=arguments, run_analysis=run_analysis, dry=dry
         )
+        if dry:
```

Comment:
Since the PR is small, could you perhaps make an extra effort to improve the code (at least a little bit). 
Suggestion:
- Add type hints to this function
- Convert `arguments` to a Pydantic object that can reveal what is expected